Decode([]byte("")) cause panic: runtime error: index out of range [3532051776] with length 0

The following cases were found during online use：

```go
func TestService_getInstalledApps(t *testing.T) {
	c, err := cuckoo.Decode([]byte(""))
	assert.Nil(t, err)
	assert.False(t, c.Lookup([]byte("test")))
}

```

output:
```
--- FAIL: TestService_getInstalledApps (0.00s)
panic: runtime error: index out of range [3532051776] with length 0 [recovered]
	panic: runtime error: index out of range [3532051776] with length 0

goroutine 19 [running]:
testing.tRunner.func1.2({0x102e36060, 0x140000e4240})
	/usr/local/go/src/testing/testing.go:1209 +0x258
testing.tRunner.func1(0x140000fe680)
	/usr/local/go/src/testing/testing.go:1212 +0x284
panic({0x102e36060, 0x140000e4240})
	/usr/local/go/src/runtime/panic.go:1038 +0x21c
```

--- a/cuckoofilter.go
+++ b/cuckoofilter.go
@@ -144,6 +144,9 @@ func Decode(bytes []byte) (*Filter, error) {
 	if len(bytes)%bucketSize != 0 {
 		return nil, fmt.Errorf("expected bytes to be multiple of %d, got %d", bucketSize, len(bytes))
 	}
+	if len(bytes) == 0 {
+		return nil, fmt.Errorf("bytes can not be empty")
+	}
 	buckets := make([]bucket, len(bytes)/4)
 	for i, b := range buckets {
 		for j := range b {

--- a/cuckoofilter_test.go
+++ b/cuckoofilter_test.go
@@ -59,6 +59,16 @@ func TestEncodeDecode(t *testing.T) {
 	}
 }
 
+func TestDecode(t *testing.T) {
+	ncf, err := Decode([]byte(""))
+	if err == nil {
+		t.Errorf("Expected err, got nil")
+	}
+	if ncf != nil {
+		t.Errorf("Expected nil, got %v", ncf)
+	}
+}
+
 func BenchmarkFilter_Reset(b *testing.B) {
 	const cap = 10000
 	filter := NewFilter(cap)

